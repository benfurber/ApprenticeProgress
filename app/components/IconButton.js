import React, { Component } from "react";
import { StyleSheet } from "react-native";
import { Icon, Text, View } from "native-base";

class IconButton extends Component {
  render() {
    const { name } = this.props;

    return (
      <View style={styles.view}>
        {this._selectIcon(name)}
        <Text style={styles.text}>{name}</Text>
      </View>
    );
  }

  _selectIcon(name) {
    if (name == "Goals") {
      return <Icon name="md-filing" style={styles.icon} />;
    }

    if (name == "Progress") {
      return <Icon name="md-speedometer" style={styles.icon} />;
    }

    return null;
  }
}

const styles = StyleSheet.create({
  icon: {},
  text: {
    paddingLeft: 10,
  },
  view: {
    flexDirection: "row",
    alignItems: "center",
  },
});

export { IconButton };
