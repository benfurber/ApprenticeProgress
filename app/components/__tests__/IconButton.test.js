import React from "react";
import { shallow } from "enzyme";

import { IconButton } from "../IconButton";

describe("IconButton", () => {
  it("has a valid snapshot", () => {
    const component = shallow(<IconButton name="Goals" />);

    expect(component).toMatchSnapshot();
  });
});
