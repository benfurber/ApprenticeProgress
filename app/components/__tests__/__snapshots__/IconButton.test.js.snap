// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconButton
    name="Goals"
    tint="#929292"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Icon)
          name="md-filing"
          style={
            Array [
              Object {},
              Object {
                "color": "#929292",
              },
            ]
          }
        />,
        <Styled(Text)
          style={
            Array [
              Object {
                "paddingLeft": 10,
              },
              Object {
                "color": "#929292",
              },
            ]
          }
        >
          Goals
        </Styled(Text)>,
      ],
      "style": Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "md-filing",
          "style": Array [
            Object {},
            Object {
              "color": "#929292",
            },
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Goals",
          "style": Array [
            Object {
              "paddingLeft": 10,
            },
            Object {
              "color": "#929292",
            },
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": "Goals",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Icon)
            name="md-filing"
            style={
              Array [
                Object {},
                Object {
                  "color": "#929292",
                },
              ]
            }
          />,
          <Styled(Text)
            style={
              Array [
                Object {
                  "paddingLeft": 10,
                },
                Object {
                  "color": "#929292",
                },
              ]
            }
          >
            Goals
          </Styled(Text)>,
        ],
        "style": Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "md-filing",
            "style": Array [
              Object {},
              Object {
                "color": "#929292",
              },
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Goals",
            "style": Array [
              Object {
                "paddingLeft": 10,
              },
              Object {
                "color": "#929292",
              },
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Goals",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
